### PR TITLE
feat: integrate reference service 0.2.0 features

### PR DESCRIPTION
**Pull request by davidlougheed**

### CODE REVIEW

**Review by gsfk — COMMENTED** [review]
Really nice, I like the tags in the autocomplete options... good enough that you might be able to remove `TODOs: style options` at the top of LocusSearch.js. The tag disappears when you select an option, not sure if that's deliberate. It might be worth keeping.

Just one thing is broken though. Since so many of our instances have only one assembly, the variants search form is supposed to autofill the "Assembly ID" field in the form if there's only one assembly available, rather than forcing you to go through a select menu with only one item. It seems like the form is not re-rendering.

**Comment by davidlougheed** [comment]
tag disappearing I don't have much of a choice on because of how the autocomplete element works / the logic that extracts the coordinates from the text itself... auto-selecting the assembly ID should be fixed now.

**Review by gsfk — APPROVED** [review]
Working now. 

Weirdly it's possible to do a custom render of the selected option in antd `Select` but not antd `Autocomplete`, despite how similar they are. This isn't the first time this component has let me down.

File: src/components/discovery/LocusSearch.js [hunk 1]
```diff
@@ -1,63 +1,55 @@
...
+const LocusSearch = ({ assemblyId, addVariantSearchValues, handleLocusChange, setLocusValidity }) => {
+    const referenceGenomes = useReferenceGenomes();
     const [autoCompleteOptions, setAutoCompleteOptions] = useState([]);
-    const geneSearchResults = useSelector((state) => state.discovery.geneNameSearchResponse);
     const [inputValue, setInputValue] = useState("");
-    const dispatch = useDispatch();
 
-    const showAutoCompleteOptions = assemblyId === "GRCh37" || assemblyId === "GRCh38";
+    const showAutoCompleteOptions = !!referenceGenomes.itemsByID[assemblyId]?.gff3_gz
+        && inputValue.length && (inputValue.includes(" ") || !inputValue.includes(":"));
```
Comment (v-rocheleau) [inline src/components/discovery/LocusSearch.js:27]: minor: useMemo

File: src/components/explorer/IndividualTracks.js [hunk 2]
```diff
@@ -161,9 +162,14 @@ const IndividualTracks = ({ individual }) => {
 
     const dispatch = useDispatch();
 
+    const referenceService = useService("reference");
     const igvGenomes = useIgvGenomes();  // Built-in igv.js genomes (with annotations)
     const referenceGenomes = useReferenceGenomes();  // Reference service genomes
 
+    const igvGenomesByID = useMemo(
```
Comment (v-rocheleau) [inline src/components/explorer/IndividualTracks.js:169]: Would it be worth it to have this in the redux store?

File: src/modules/reference/hooks.js [hunk 3]
```diff
@@ -12,3 +13,49 @@ export const useReferenceGenomes = () => {
...
+        const params = new URLSearchParams({ name: nameQuery, name_fzy: "true", limit: "10" });
+        const searchUrl = `${referenceService.url}/genomes/${referenceGenomeID}/features?${params.toString()}`;
+
+        setError(null);
+
+        (async () => {
+            setIsFetching(true);
+
+            try {
+                const res = await fetch(searchUrl, { headers: { Accept: "application/json" } });
```
Comment (v-rocheleau) [inline src/modules/reference/hooks.js:42]: Are we sure the reference service will always be public? If not, we should pass the auth headers as well.

File: src/components/discovery/LocusSearch.js [hunk 4]
```diff
@@ -1,63 +1,55 @@
...
-    const handleChange = (value) => {
+        addVariantSearchValues({ chrom, start, end });
+    }, [setLocusValidity, addVariantSearchValues]);
 
-        setInputValue(value);
-
-        if (!value.includes(" ") && value.includes(":")) {
-            handlePositionNotation(value);
+    useEffect(() => {
+        if (!inputValue.includes(" ") && inputValue.includes(":")) {
```
Comment (v-rocheleau) [inline src/components/discovery/LocusSearch.js:36]: minor: could factor out `inputValue.includes(" ")`, `inputValue.includes(":")` into constants for reuse here and to compute `showAutoCompleteOptions`

**Review by v-rocheleau — CHANGES_REQUESTED** [review]
looks good, small comments

**Review by v-rocheleau — APPROVED** [review]
lgtm!

**Pull request merged**